code/datums/weather/weather_types/snow_storm.dm

```diff
@@ -27,3 +27,25 @@
 /datum/weather/snow_storm/weather_act(mob/living/L)
 	L.adjust_bodytemperature(-rand(5,15))
 
+
+// since snowstorm is on a station z level, add extra checks to not annoy everyone
+/datum/weather/snow_storm/can_get_alert(mob/player)
+	var/standard_check = ..()
+	// dont bother checking if theyre not on the station or wouldnt be able to see the alert anyways
+	if(!standard_check || !is_station_level(player.z))
+		return standard_check
+
+	if(isobserver(player))
+		return TRUE
+
+	if(istype(get_area(player), /area/mine))
+		return TRUE
+
+	if(player.mind?.assigned_role.title == JOB_SHAFT_MINER) // a good miner always senses a storm brewin.
```

Comment:
This needs to be either a mind trait or a skillchip. We don't want any more explicit job checks like this.

Reply:
Also this comment is useless